src/__tests__/Search/SearchResultsList.test.tsx

```diff
@@ -74,55 +68,40 @@ describe('SearchResultsList', () => {
   });
 
   it('should clear the checkbox when a checked result is clicked', async () => {
-    const { testData } = getTestData();
-    global.fetch = jest.fn(() =>
-      Promise.resolve({
-        json: () => ({
-          results: testData,
-        }),
-      }),
-    ) as jest.Mock;
-    jest.spyOn(global, 'fetch');
     // set delay to null to prevent test time-out due to useFakeTimers
     const user = userEvent.setup({ delay: null });
 
     renderComponent();
-    // focus input to show results
 
+    // focus input to show results
     const searchInput = screen.getAllByRole('textbox')[0];
     await user.click(searchInput);
 
-    const fleshWound = await screen.findAllByText("it's just a flesh wound");
-
-    await user.click(fleshWound[0]);
-    act(() => {
-      expect(store.getState().search[searchType].checkedRevisions[0]).toBe(
-        testData[1],
-      );
+    const fleshWound = await screen.findByRole('button', {
+      name: /it's just a flesh wound/,
     });
+    const fleshWoundCheckbox = within(fleshWound).getByRole('checkbox');
 
-    await user.click(fleshWound[0]);
-    expect(fleshWound[0].classList.contains('Mui-checked')).toBe(false);
+    await user.click(fleshWound);
+    expect(fleshWound).toHaveClass('item-selected');
+    expect(fleshWoundCheckbox).toBeChecked();
+    expect(fleshWound.querySelector('.Mui-checked')).toBeInTheDocument();
+
+    await user.click(fleshWound);
+    expect(fleshWound).not.toHaveClass('item-selected');
+    expect(fleshWoundCheckbox).not.toBeChecked();
+    expect(fleshWound.querySelector('.Mui-checked')).toBeNull();
```

Comment:
I rewrote the expectations extensively, so that it doesn't check internal state.